src/AgentMulder.ReSharper.Domain/Patterns/INavigationProvider.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.CSharp.Tree;
+
+namespace AgentMulder.ReSharper.Domain.Patterns
+{
+    public interface INavigationProvider
```

Comment:
Name feels too generic. I'd say this is something that provides navigation services. Can we think of something more specific?

Reply:
Not sure how could I name it. What would you say for IGotoInstantiationNavigationProviderExtensions or INavigationProviderExtensions ? In the future, there is a possibility to use this interface in GoToRegisteredComponentsNavigationProvider.cs